core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -563,6 +606,21 @@ object StorageTool extends Logging {
         })
       })
       copier.writeLogDirChanges()
+      if (advertisedListenerEndpoints.nonEmpty) {
+        metaPropertiesEnsemble.emptyLogDirs().forEach(logDir => {
+          writeCheckpointFile(stream, logDir, copier.logDirProps().get(logDir), advertisedListenerEndpoints)
+        })
+      }else if (controllersQuorumVoters != null) {
+        metaPropertiesEnsemble.emptyLogDirs().forEach(logDir => {
+          val nodeId = copier.logDirProps().get(logDir).nodeId()
+          val voterMap: util.Map[Integer, InetSocketAddress] = parseVoterConnections(Collections.singletonList(controllersQuorumVoters))
+          voterMap.keySet().forEach(replicaId => {
+            if (nodeId.getAsInt == replicaId){
+              //writeCheckpointFile(stream, logDir, copier.logDirProps().get(logDir), voterMap)
```

Comment:
@jsancio does this make sense to write the checkpoint file if this condition is true ?